apps/cyberstorm-nextjs/middleware.ts

```diff
@@ -0,0 +1,56 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import {
+  decode,
+  handleJWTGenerationAndReturnResponseOrRedirect,
+  redirectToPremium,
+} from "./utils/auth";
+import { errors } from "jose";
+
+export const config = {
+  matcher: [
+    /*
+     * Match all request paths except for the ones starting with:
+     * - _next/static (static files)
+     * - _next/image (image optimization files)
+     * - favicon.ico (favicon file)
+     */
+    "/((?!_next/static|_next/image|favicon.ico|images).*)",
+  ],
+};
+
+export async function middleware(request: NextRequest) {
+  const req = request;
+  const response = NextResponse.next();
+
+  // For the sake of keeping dev configuration easy and not confusing
+  if (process.env.CYBERSTORM_AUTH_ENABLED) {
+    return response;
+  }
+
+  // JWT exists?
+  if (request.cookies.has("CyberstormAuth")) {
+    // Validate JWT
+
```

Comment:
Nitpick but I'm irked about the empty line after this comment. Also I'm not 100% sure that the comments in this if-block add that much value, they mostly just say the same thing the code does?

Reply:
Now that I think about; true. I'll remove all of those that, give no value 👍